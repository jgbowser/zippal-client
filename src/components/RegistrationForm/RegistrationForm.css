.RegistrationForm__link {
  color: black;
  font-weight: 500;
}

.RegistrationForm__required_text {
  margin-top: 0;
  margin-bottom: 25px;
  font-size: 12px;
}