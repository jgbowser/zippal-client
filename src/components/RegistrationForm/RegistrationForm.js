import React, {Component} from 'react'
import {Link} from 'react-router-dom'
import AuthApiService from '../../services/auth-api-service'
import UserContext from '../../context/UserContext';
import Required from '../Required/Required'


class RegistrationForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => { }
  }

  state = {error:null}

  static contextType = UserContext

  firstInput = React.createRef()

  handleSubmit = ev => {
    ev.preventDefault()
    const {name,username,password} = ev.target
    AuthApiService.postUser({
      display_name: name.value,
      username: username.value,
      password: password.value,
    })
    .then(() => {
      AuthApiService.postLogin({
        username: username.value,
        password: password.value
      }).then((res) => {
        this.context.processLogin(res.authToken);
      }).then(user => {
        name.value = ''
        username.value = ''
        password.value = ''
        this.props.onRegistrationSuccess()
      })
    })
    .catch(res => {
      this.setState({error: res.error})
    })
  }

  componentDidMount(){
    this.firstInput.current.focus()
  }

  render(){
    const {error} = this.state
    return (
      <form
      className='user_form'
        onSubmit={this.handleSubmit}
      >
        <legend>
        <div role='alert'>
          {error && <p className='error'>{error}</p>}
        </div>
        <p><Required /> indicates required field</p>
        <div>
          <label htmlFor='registration-name-input'>
            Display Name<Required />
          </label>
          <input
            ref={this.firstInput}
            id='registraion-name-input'
            name='name'
            required
          />
        </div>
        <div>
          <label htmlFor='registration-username-input'>
            Username<Required />
          </label>
          <input
            id='registration-username-input'
            name='username'
            required
          />
        </div>
        <div>
          <label htmlFor='registration-password-input'>
            Password<Required />
          </label>
          <input
            id='registration-password-input'
            name='password'
            type='password'
            required
          />
        </div>
        <footer>
          <button type='submit'>
            Sign up
          </button>
          {' '}
          <Link to='/login'>Already have an account?</Link>
        </footer>
        </legend>
      </form>
    )
  }
}

export default RegistrationForm