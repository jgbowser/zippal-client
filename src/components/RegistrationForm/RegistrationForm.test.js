import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import RegistrationForm from './RegistrationForm'
import renderer from 'react-test-renderer'

describe('RegistrationForm Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <RegistrationForm />
      </BrowserRouter>, div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  it(`takes a snapshot of the RegistrationForm Component`,()=>{
    const tree = renderer
    .create(<BrowserRouter><RegistrationForm/></BrowserRouter>)
    expect(tree).toMatchSnapshot()
  })
})