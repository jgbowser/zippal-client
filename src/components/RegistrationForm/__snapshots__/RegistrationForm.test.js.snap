// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm Component takes a snapshot of the RegistrationForm Component 1`] = `
<form
  className="user_form"
  onSubmit={[Function]}
>
  <legend>
    <div
      role="alert"
    />
    <p
      className="RegistrationForm__required_text"
    >
      <span
        className="required"
      >
        *
      </span>
       indicates required field
    </p>
    <div>
      <label
        htmlFor="registration-name-input"
      >
        Display Name
        <span
          className="required"
        >
          *
        </span>
      </label>
      <input
        id="registration-name-input"
        name="name"
        required={true}
      />
    </div>
    <div>
      <label
        htmlFor="registration-username-input"
      >
        Username
        <span
          className="required"
        >
          *
        </span>
      </label>
      <input
        id="registration-username-input"
        name="username"
        required={true}
      />
    </div>
    <div>
      <label
        htmlFor="registration-password-input"
      >
        Password
        <span
          className="required"
        >
          *
        </span>
      </label>
      <input
        id="registration-password-input"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <div>
      <button
        className="form_button"
        type="submit"
      >
        Sign up
      </button>
       
      <a
        className="RegistrationForm__link"
        href="/login"
        onClick={[Function]}
      >
        Already have an account?
      </a>
    </div>
  </legend>
</form>
`;
